packages/dev/core/src/Rendering/GaussianSplatting/gaussianSplatting.ts

```diff
@@ -284,36 +310,47 @@ export class GaussianSplatting {
             this.dispose();
         }
         this._setData(new Uint8Array(data as any));
-        const matricesData = new Float32Array(this.vertexCount * 16);
+        const matricesData = new Float32Array(this.vertexCount * 1); // matrix is only used to allocate instances
+        const splatIndex = new Float32Array(this.vertexCount * 1);
 
         const updateInstances = (indexMix: Uint32Array) => {
             for (let j = 0; j < this.vertexCount; j++) {
-                const i = indexMix[2 * j];
-                const index = j * 16;
-                matricesData[index + 0] = this._positions[i * 3 + 0];
-                matricesData[index + 1] = this._positions[i * 3 + 1];
-                matricesData[index + 2] = this._positions[i * 3 + 2];
-
-                matricesData[index + 4] = this._uBuffer[32 * i + 24 + 0] / 255;
-                matricesData[index + 5] = this._uBuffer[32 * i + 24 + 1] / 255;
-                matricesData[index + 6] = this._uBuffer[32 * i + 24 + 2] / 255;
-                matricesData[index + 7] = this._uBuffer[32 * i + 24 + 3] / 255;
-
-                matricesData[index + 8] = this._covA[i * 3 + 0];
-                matricesData[index + 9] = this._covA[i * 3 + 1];
-                matricesData[index + 10] = this._covA[i * 3 + 2];
-
-                matricesData[index + 12] = this._covB[i * 3 + 0];
-                matricesData[index + 13] = this._covB[i * 3 + 1];
-                matricesData[index + 14] = this._covB[i * 3 + 2];
+                splatIndex[j] = indexMix[2 * j];
             }
-
-            this.mesh?.thinInstanceBufferUpdated("matrix");
+            this.mesh?.thinInstanceBufferUpdated("splatIndex"); // update splatIndex only
         };
 
         // update so this.mesh is valid when exiting this function
         this.mesh = this._getMesh(this.scene);
-        this.mesh.thinInstanceSetBuffer("matrix", matricesData, 16, false);
+        this.mesh.thinInstanceSetBuffer("matrix", matricesData, 1, true);
+        this.mesh.thinInstanceSetBuffer("splatIndex", splatIndex, 1, false);
+
+        /// create textures for gaussian info
+        if (this._material.name == "GaussianSplattingShader") {
+            let material = this.mesh.material as ShaderMaterial;
+
+            let textureSize = this.getTextureSize(this.vertexCount);
+            material.setVector2("dataTextureSize", textureSize);
+
+            let convATexture = new RawTexture(this._covA, textureSize.x, textureSize.y, Engine.TEXTUREFORMAT_RGB, this.scene, false, false, Engine.TEXTURE_BILINEAR_SAMPLINGMODE, Engine.TEXTURETYPE_FLOAT); // floating issue

```

Comment:
RGB is not supported by WebGPU.

Anyway, for maximum performance, I think you should do as in https://github.com/mkkellogg/GaussianSplats3D/blob/main/src/SplatMesh.js:
* have a single texture for covA and covB. Either RG16Float if supported (`engine.getCaps().textureHalfFloat === true`), otherwise RG32Float
* have a single texture for center + color. RGBA32Uint should do the trick, with `xyz` for center and `a` for color. Encoding the color in 32 bits (8 bits per component) is fine, we shouldn't have any visible differences with the use of a half or full float. See how the packing/unpacking of center/color is done in the provided link.

Let's see what @CedricGuillemet think about it.

Reply:
Thank you~ @Popov72, compressing the four textures into two is a fantastic idea. Do you think it's necessary to merge the current four-texture version? Texture compression sounds like a significant change, and I might not be able to implement it recently. If the current version is acceptable, I can proceed with updating other review changes and attempt to merge them.

Reply:
I like smal iterations. I'm fine with doing these changes in a future PR.But in the mean time, it must work with webgl and webgpu.

Reply:
@CedricGuillemet Got it, looks like the orignal shader does not compile in webgl & webgpu context.
With webgl there is no `transpose` function, I can add one in shader.
With webgpu there is a [non-opaque issue](https://forum.babylonjs.com/t/webgpu-use-glsl-bug-non-opaque-uniforms-outside-a-block/42972) issue, should I add a different version of shader to support webgpu? Or is the shader supposed to be processed by the engine?

Reply:
does your work with GaussianMaterial will fix this issue @sebavan  ?

Reply:
Not really, let see if @Popov72 has some ideas ?

Reply:
I don't understand what is the "non-opaque issue", as the linked thread has it resolved?

I'm fine with merging the current PR and improving in another iteration. You should just use a RGBA texture instead of RGB to make it work in WebGPU.

Reply:
To fix the "non opaque" problem, you should replace:
```glsl
uniform mat4 projection, modelView;
```
with:
```glsl
uniform mat4 projection;
uniform mat4 modelView;
```
For performance reasons of the shader code processing, we don't support multiple definitions of uniforms in a single line (also true for varying / attribute).

Reply:
> To fix the "non opaque" problem, you should replace:
> 
> ```glsl
> uniform mat4 projection, modelView;
> ```
> 
> with:
> 
> ```glsl
> uniform mat4 projection;
> uniform mat4 modelView;
> ```
> 
> For performance reasons of the shader code processing, we don't support multiple definitions of uniforms in a single line (also true for varying / attribute).

thanks! after some other changes it now works with webgl and webgpu in my playground